paasta_tools/paasta_cli/cmds/validate.py

```diff
@@ -0,0 +1,120 @@
+    "http://paasta.readthedocs.org/en/latest/yelpsoa_configs.html")
+
+SCHEMA_NOT_FOUND = failure(
+    "Failed to find schema to validate against. More info:",
+    "http://paasta.readthedocs.org/en/latest/yelpsoa_configs.html")
+
+FAILED_READING_FILE = failure(
+    "Failed to read file. More info:",
+    "http://paasta.readthedocs.org/en/latest/yelpsoa_configs.html")
+
+
+def get_schema(file_type):
+    """Get the correct schema to use for validation
+
+    :param file_type: what schema type should we validate against
+    """
+    schema_path = 'schemas/%s_schema.json' % file_type
+    try:
+        schema = pkgutil.get_data('paasta_tools.paasta_cli', schema_path)
+    except IOError:
+        return None
+    return json.loads(schema)
+
+
+def validate_schema(file_path, file_type):
+    """Check if the specified config file has a valid schema
+
+    :param file_path: path to file to validate
+    :param file_type: what schema type should we validate against
+    """
+    schema = get_schema(file_type)
+    if (schema is None):
+        print '%s: %s' % (SCHEMA_NOT_FOUND, file_path)
+        return
+    validator = Draft4Validator(schema, format_checker=FormatChecker())
+    basename = os.path.basename(file_path)
+    extension = os.path.splitext(basename)[1]
+    try:
+        config_file = get_file_contents(file_path)
+    except IOError:
+        print '%s: %s' % (FAILED_READING_FILE, file_path)
+        return
+    if extension == '.yaml':
+        config_file_object = yaml.load(config_file)
+    elif extension == '.json':
+        config_file_object = json.loads(config_file)
+    else:
+        config_file_object = config_file
+    try:
+        validator.validate(config_file_object)
+    except ValidationError as e:
+        print '%s: %s' % (SCHEMA_INVALID, file_path)
+        print '  Validation Message: %s' % e.message
+    else:
+        print '%s: %s' % (SCHEMA_VALID, file_path)
+
+
+def validate_all_schemas(service_path):
+    """Finds all recognized config files in service directory,
+    and validates their schema.
+
+    :param service_path: path to location of configuration files"""
+
+    path = os.path.join(service_path, '*.yaml')
+
+    for file_name in glob(path):
+        basename = os.path.basename(file_name)
+        for file_type in ['chronos', 'marathon']:
+            if basename.startswith(file_type):
+                validate_schema(basename, file_type)
+
+
+def add_subparser(subparsers):
+    validate_parser = subparsers.add_parser(
+        'validate',
+        description="Execute 'paasta validate' from service repo root",
+        help="Validate that all paasta config files in pwd are correct")
+    validate_parser.set_defaults(command=paasta_validate)
+
+
+def paasta_validate(args):
+    """Analyze the service in the PWD to determine if conf files are all valid
+    :param args: argparse.Namespace obj created from sys.args by paasta_cli"""
+    service_path = os.getcwd()
```

Comment:
I still fail to see the use case for this. The workflow I envisioned was a user is editing their service's soa configs and wants to make sure they are valid. So they run 'paasta validate', make any necessary changes, and repeat. This workflow does not require anything to be committed or any waiting.

The only scripts in paasta_tools/paasta_cli/cmds that actually support specifying yelpsoa-config-root are cook_image, emergency_scale, fsm, and local_run. Notice that `paasta check`, which is quite similar to `paasta validate` is not in this list.

Since it is pretty clear that the rest of the team feels this change would be useful, I will make it. However, I would love it if someone could describe in a bit more detail how they see the --soa-dir argument being used in practice and why this differs from the `paasta check` workflow.